src/components/Heading/src/Heading.vue

```diff
@@ -90,9 +100,19 @@ export default {
 			return `minus${this.resolvedSize}`;
 		},
 		inlineStyles() {
+			const { fonts } = this.theme;
 			return {
 				fontFamily: this.resolvedFontFamily,
 				color: this.resolvedTextColor,
+				fontWeight: this.resolvedWeight,
+				'--mobile-base-font-size': fonts.mobileBaseSize,
+				'--mobile-font-size-scale': fonts.mobileSizeScale,
+				'--mobile-base-line-height': fonts.mobileBaseLineHeight,
+				'--mobile-line-height-scale': fonts.mobileLineHeightScale,
+				'--desktop-base-font-size': fonts.desktopBaseSize,
+				'--desktop-font-size-scale': fonts.desktopSizeScale,
+				'--desktop-base-line-height': fonts.desktopBaseLineHeight,
+				'--desktop-line-height-scale': fonts.desktopLineHeightScale,
```

Comment:
We need a better way of globally setting these values so they can be shared across multiple components. It's inefficient and costly to performance to set so many values inline.

I recommend we move all the css properties that can be set globally across multiple components into one source and finally tackle this issue: https://github.com/square/maker/issues/39

Reply:
> We need a better way of globally setting these values so they can be shared across multiple components.

We already have a way, it's called the MTheme component.

> It's inefficient and costly to performance to set so many values inline.

I think the performance cost is negligible.

> I recommend we move all the css properties that can be set globally across multiple components into one source

We already do that, it's called the MTheme component.

> and finally tackle this issue: #39

That issue only concerns static CSS. Even if that issue is resolved it would not remove the inline styles used in the MHeading and MText components which are for run-time customizable dynamic CSS and required to be there if the components are meant to be usable without a parent MTheme component.